### PR TITLE
Renamed master key to principal key

### PR DESCRIPTION
**Pull request by dutow**
This commit contains lots of changes, but it's just a repeated
execution of find <...> -exec sed <...>, so everything should
work as before.

Note: PR also contains changes from the SMGR branch merge, as it is based on that, please only check the latest commit: 9c08d95c9b4d4a6ce79f6a42941f5d7ddaa65aa6

### CODE REVIEW

File: src/catalog/tde_global_catalog.c [hunk 1]
```diff
@@ -215,7 +215,7 @@ init_gl_catalog_keys(void)
 }
 
 static TDEMasterKey *
-create_master_key(const char *key_name, GenericKeyring * keyring,
+create_principal_key(const char *key_name, GenericKeyring * keyring,
 				  Oid dbOid, Oid spcOid, bool ensure_new_key)
 {
 	TDEMasterKey *masterKey;
```
Comment (dAdAbird) [inline src/catalog/tde_global_catalog.c:221]: The variable name (along with the type)

File: src/catalog/tde_principal_key.c [hunk 2]
```diff
@@ -51,29 +51,29 @@ typedef struct TdeMasterKeylocalState
...
-/* parameter for the master key info shared hash */
-static dshash_parameters master_key_dsh_params = {
+/* parameter for the principal key info shared hash */
+static dshash_parameters principal_key_dsh_params = {
     sizeof(Oid),
     sizeof(TDEMasterKey),
     dshash_memcmp, /* TODO use int compare instead */
     dshash_memhash};
 
 TdeMasterKeylocalState masterKeyLocalState;
```
Comment (dAdAbird) [inline src/catalog/tde_principal_key.c:61]: Both the type and variable

**Review by dAdAbird — CHANGES_REQUESTED** [review]
I made a couple of comments marking not renamed entities. But there are quite a bit of them.
I think patterns *MasterKey* and *masterKey* were omitted.
Plus there is a variable  `mastere_key_info` (a misspelling and a snake case) in [load_latest_versioned_key_name()](https://github.com/Percona-Lab/pg_tde/pull/228/commits/853a923b695ee7469d18ed84cdbd7c106624ee30#diff-fc8109ce322645740f720cb5a9101e7aef73c26a54354df2abc93d60d5007111R510)  (src/catalog/tde_principal_key.c)

**Comment by dutow** [comment]
@dAdAbird yes, I only searched for master.key, not [mM]aster[k]Key. Fixed now.

File: src/catalog/tde_principal_key.c [hunk 3]
```diff
@@ -491,23 +491,23 @@ RotateMasterKey(const char *new_key_name, const char *new_provider_name, bool en
...
 
 /*
-* Load the latest versioned key name for the master key
+* Load the latest versioned key name for the principal key
 * If ensure_new_key is true, then we will keep on incrementing the version number
 * till we get a key name that is not present in the keyring
 */
 keyInfo *
-load_latest_versioned_key_name(TDEMasterKeyInfo *mastere_key_info, GenericKeyring *keyring, bool ensure_new_key)
+load_latest_versioned_key_name(TDEPrincipalKeyInfo *mastere_key_info, GenericKeyring *keyring, bool ensure_new_key)
```
Comment (dAdAbird) [inline src/catalog/tde_principal_key.c:510]: `mastere_key_info` here and all over the func body + in the function prototype, in `src/include/catalog/tde_principal_key.h`

**Review by dAdAbird — CHANGES_REQUESTED** [review]
Found a couple more

**Review by dAdAbird — APPROVED** [review]

**Comment by Adammulla** [comment]
Hello Team, When I try to install and configure TDE I am getting create_principal_key don't exists. 
I checked in documentation principal key used instead of master key earlier it was master key in documentation but now changed to principal key . can you please set this in setup code also?

**Pull request merged**